src/pages/AddItemPage.tsx

```diff
@@ -0,0 +1,151 @@
+import { useEffect, useState, useRef } from "react";
+import FileInput from "../Component/FileInput";
+import "../../src/style/AddItemPage.css";
+
+interface AddItemState {
+  imgFile: File | null;
+  title: string;
+  description: string;
+  price: string;
+  tag: string[];
+}
+
+function AddItemPage() {
+  const [addItemValue, setAddItemValue] = useState<AddItemState>({
+    imgFile: null,
+    title: "",
+    description: "",
+    price: "",
+    tag: [],
+  });
+
+  const handleChange = (name: string, value: string | File | null) => {
+    setAddItemValue((prevValues) => ({
+      ...prevValues,
+      [name]: value,
+    }));
+  };
+
+  const enterEvent = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === "Enter" && e.currentTarget.value.trim() !== "") {
+      handleTagChange(e.currentTarget.value.trim());
+      e.currentTarget.value = "";
+    }
+  };
+
+  const handleTagChange = (newTag: string) => {
+    setAddItemValue((prevValues) => ({
+      ...prevValues,
+      tag: [...prevValues.tag, newTag],
+    }));
+    console.log(addItemValue);
+  };
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = e.currentTarget;
+    handleChange(name, value);
+  };
+
+  const deleteTag = (deleteTag: string) => {
+    setAddItemValue((prevValues) => ({
+      ...prevValues,
+      tag: prevValues.tag.filter((tag) => tag !== deleteTag),
+    }));
+  };
+
+  const submitButton = useRef<HTMLButtonElement>(null);
+  const [buttonDisabled, setButtonDisabled] = useState(true);
+
+  const test = () => {
+    console.log("등록성공");
+  };
+
+  useEffect(() => {
+    const buttonOnOff = () => {
+      return !(
+        addItemValue.title !== "" &&
+        addItemValue.description !== "" &&
+        addItemValue.price !== "" &&
+        addItemValue.tag.length > 0
+      );
+    };
```

Comment:
이건 아에 useMemo로 빼는게 더 좋겠는데요?ㅎㅎ

```suggestion
    const canSubmit = useMemo<boolean>(() => {
      return !(
        addItemValue.title !== "" &&
        addItemValue.description !== "" &&
        addItemValue.price !== "" &&
        addItemValue.tag.length > 0
      );
    }, [addItemValue]);
```

Reply:
아 모르는 hook이었습니다. 알겠습니다!

Reply:
앗앗?? 혹시 useMemo랑 useCallback 배우지 않으셨었나요?
몰라도 지금당장에는 문제는 없긴 한데... 꽤 중요하고 자주 사용될꺼라ㅠ 성능적인 이점을 가져가기 위해서 필요해요...! 

한번 조사해보시면 좋을것같아요~!

Reply:
강의에 있었을지도 모르긴하는데 아직 잘 생각이안나서 그냥 어떻게든 진행하는 경우가 있어서 그렇습니다!